cmd/soroban-rpc/internal/db/event.go

```diff
@@ -97,3 +112,225 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
+				OrderBy("ledger_sequence DESC").
+				Limit(1),
+			"m2",
+		).ToSql()
+	if err != nil {
+		return ledgerRange, fmt.Errorf("couldn't build ledger range query: %w", err)
+	}
+
+	// Combine the min and max ledger sequence queries using UNION ALL
+	eventMinMaxLedgersQ, _, err := minLedgerSeqQ.Suffix("UNION ALL "+maxLedgerSeqQ, args...).ToSql()
+	if err != nil {
+		return ledgerRange, fmt.Errorf("couldn't build ledger range query: %w", err)
+	}
+
+	// Final query to join ledger_close_meta table and the sequence numbers we got from eventMinMaxLedgersQ
+	finalSQL := sq.
+		Select("lcm.meta").
+		From(ledgerCloseMetaTableName + " as lcm").
+		JoinClause(fmt.Sprintf("JOIN (%s) as seqs ON lcm.sequence == seqs.ledger_sequence", eventMinMaxLedgersQ))
+
+	var lcms []xdr.LedgerCloseMeta
+	if err = eventHandler.db.Select(ctx, &lcms, finalSQL); err != nil {
+		return ledgerRange, fmt.Errorf("couldn't build ledger range query: %w", err)
+	} else if len(lcms) < 2 {
+		// There is almost certainly a row, but we want to avoid a race condition
+		// with ingestion as well as support test cases from an empty DB, so we need
+		// to sanity check that there is in fact a result. Note that no ledgers in
+		// the database isn't an error, it's just an empty range.
+		return ledgerRange, nil
+	}
+
+	lcm1, lcm2 := lcms[0], lcms[1]
+	ledgerRange.FirstLedger.Sequence = lcm1.LedgerSequence()
+	ledgerRange.FirstLedger.CloseTime = lcm1.LedgerCloseTime()
+	ledgerRange.LastLedger.Sequence = lcm2.LedgerSequence()
+	ledgerRange.LastLedger.CloseTime = lcm2.LedgerCloseTime()
+
+	eventHandler.log.Debugf("Database ledger range: [%d, %d]",
+		ledgerRange.FirstLedger.Sequence, ledgerRange.LastLedger.Sequence)
+	return ledgerRange, nil
+}
+
+type eventTableMigration struct {
+	firstLedger uint32
+	lastLedger  uint32
+	writer      EventWriter
+}
+
+func (e *eventTableMigration) ApplicableRange() *LedgerSeqRange {
+	return &LedgerSeqRange{
+		firstLedgerSeq: e.firstLedger,
+		lastLedgerSeq:  e.lastLedger,
+	}
+}
+
+func (e *eventTableMigration) Apply(_ context.Context, meta xdr.LedgerCloseMeta) error {
+	return e.writer.InsertEvents(meta)
+}
+
+func newEventTableMigration(
+	ctx context.Context,
+	logger *log.Entry,
+	retentionWindow uint32,
+	passphrase string,
+) migrationApplierFactory {
+	return migrationApplierFactoryF(func(db *DB, latestLedger uint32) (MigrationApplier, error) {
+		firstLedgerToMigrate := uint32(2)
+		writer := &eventHandler{
+			log:        logger,
+			db:         db,
+			stmtCache:  sq.NewStmtCache(db.GetTx()),
+			passphrase: passphrase,
+		}
+		if latestLedger > retentionWindow {
+			firstLedgerToMigrate = latestLedger - retentionWindow
+		}
+
+		// Truncate the table, since it may contain data, causing insert conflicts later on.
+		_, err := db.Exec(ctx, sq.Delete(transactionTableName).Where(sq.Lt{"ledger_sequence": firstLedgerToMigrate}))
+		if err != nil {
+			return nil, fmt.Errorf("couldn't truncate the table %q: %w", transactionTableName, err)
+		}
```

Comment:
You don't need to do this since the migration is being applied on an empty DB. In the case of transactions we realized there was no migration after the code was in the wild.